test/src/test/java/hudson/model/DescriptorVisibilityFilterTest.java

```diff
@@ -43,6 +52,67 @@ public void jenkins40545() throws Exception {
         assertThat(page.getWebResponse().getContentAsString(), containsString("descriptors found: .")); // No output written from expression
     }
 
+    @Test @Issue("JENKINS-49044")
+    public void securityRealmAndAuthStrategyHidden() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(AuthorizationStrategy.UNSECURED);
+        HtmlPage page = j.createWebClient().goTo("configureSecurity");
+        String response = page.getWebResponse().getContentAsString();
+        assertThat(response, not(containsString("TestSecurityRealm")));
```

Comment:
More convincing for the `DescriptorVisibilityFilter` impls (which BTW could be consolidated to just one) to be dynamic, so your test can first visit `/configureSecurity`, prove that they are displayed as a control, then flip the switch, visit it again, and prove that they are hidden.

Otherwise your test is weak since it would continue to pass if you, say, dropped all the `@TestExtension` annotations.